scrapers/scraper.py

```diff
@@ -0,0 +1,434 @@
+            os.path.join(path, "scrapers", dirName, ".mailmap"),
+        )
+        lines = subprocess.check_output(
+            [
+                "git",
+                "log",
+                "--format=%aN",
+                "--follow",
+                "apertium-{0}.{0}.lexc".format(language),
+            ],
+            cwd=os.path.join(path, "scrapers", dirName),
+        ).decode("utf-8")
+
+        if updatemailmap:
+            emails = (
+                subprocess.check_output(
+                    [
+                        "git",
+                        "log",
+                        "--format=<%aE>",
+                        "--follow",
+                        "apertium-{0}.{0}.lexc".format(language),
+                    ],
+                    cwd=os.path.join(path, "scrapers", dirName),
+                )
+                .decode("utf-8")
+                .split("\n")
+            )
+            mailmap = open(
+                os.path.join(path, "scrapers", ".mailmap"), "r", encoding="utf-8"
+            ).read()
+            for email in emails:
+                if not email in mailmap:
+                    print(email)
+
+        authors = lines.split("\n")
+        authors.pop()  # last line is always empty
+        contributors = []
+        authorCount = Counter(authors)
+        for contributor, count in authorCount.items():
+            contributors.append({"user": contributor, "value": count})
+
+        wikiData = requests.get(wikiURL).text
+        rows = etree.fromstring(wikiData, parser=etree.XMLParser(encoding="utf-8"))[1][
+            0
+        ][0][0][3][3][8]
+        for row in rows[2:]:
+            if row[0][0][0].text[len("apertium-") :] == language:
+                state = row[6].text.strip()
+                break
+            state = "unknown"
+
+        data.append(
+            {
+                "lang": language,
+                "state": state,
+                "stems": stems,
+                "location": "{} ({})".format(dirName, location),
+                "contributors": contributors,
+            }
+        )
+
+    return data
+
+
+def pairData(packages):
+    """Returns the locations and stems of all specified pairs"""
+    data = []
+    for package in packages:
+        if not re.match(r"apertium-\w+-\w+", package["name"]):
+            continue
+
+        pairName = package["name"].replace("apertium-", "")
+        pairSet = set(pairName.split("-"))
+        if not pairSet <= set(languages):
+            continue
+        for topic in package["topics"]:
+            if topic[len("apertium-") :] in pairLocations:
+                location = topic[len("apertium-") :]
+                break
+        stems = json.loads(
+            requests.get(
+                "https://apertium.projectjj.com/stats-service/apertium-{}/bidix".format(
+                    pairName
+                )
+            ).text
+        )["stats"][0]["value"]
```

Comment:
Does getting the `0`th stat always work?

Reply:
Yes. It return a single element list

Reply:
stats-service can return as many stats as it wants though. It just happens to return one for this particular file-type right now: https://github.com/apertium/apertium-stats-service/blob/2cde9217598a19d5e13ac0deb72fe872b19a7874/src/models.rs#L11. I'd suggest making this code robust to the service returning more things.